File: leetcode/49. Group Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> m;
        for(auto& str : strs){
            string key = str;
            sort(key.begin(), key.end());
            m[key].push_back(std::move(str));
        }
        
        vector<vector<string>> res;
        for(auto iter = m.begin(); iter != m.end(); ++iter){
            res.push_back(std::move(iter->second));
        }
        return res;
    }
};

File: leetcode/49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = {}
        
        for s in strs:
            chars = [0 for _ in range(26)]
            for c in s:
                chars[ord(c) - ord('a')] += 1
            chars = tuple(chars)
            if chars not in dic:
                dic[chars] = []
            dic[chars].append(s)
        
        return list(dic.values())
